components/button/floating-buttons.js

```diff
@@ -0,0 +1,251 @@
+			}
+			:host-context([dir="rtl"]) .d2l-floating-buttons-inner-container ::slotted(d2l-button),
+			:host-context([dir="rtl"]) .d2l-floating-buttons-inner-container ::slotted(button),
+			:host-context([dir="rtl"]) .d2l-floating-buttons-inner-container ::slotted(.d2l-button) {
+				margin-left: 0.75rem !important;
+				margin-right: 0 !important;
+			}
+
+			@keyframes d2l-floating-buttons-animation {
+				0% {
+					border-color: transparent;
+					background-color: transparent;
+					transform: translate(0,10px);
+				}
+				100% {
+					border-top-color: var(--d2l-color-mica);
+					background-color: rgba(255, 255, 255, 0.88);
+					transform: translate(0,0);
+				}
+			}
+			@-webkit-keyframes d2l-floating-buttons-animation {
+				0% {
+					border-color: transparent;
+					background-color: transparent;
+					-webkit-transform: translate(0,10px);
+				}
+				100% {
+					border-top-color: var(--d2l-color-mica);
+					background-color: rgba(255, 255, 255, 0.88);
+					-webkit-transform: translate(0,0);
+				}
+			}
+			`
+		];
+	}
+
+	constructor() {
+		super();
+		this.minHeight = '500px';
+		this._buttonSpacerStyle = {};
+		this._container = null;
+		this._containerClasses = {
+			'd2l-floating-buttons-container': true,
+			'd2l-floating-buttons-floating': this._floatingButtonsFloating
+		};
+		this._floatingButtonsFloating = false;
+		this._innerContainerStyle = {};
+		this._reposition = this._reposition.bind(this);
+		this._spacer = null;
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		window.addEventListener('resize', this._reposition);
+		window.addEventListener('scroll', this._reposition);
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		window.removeEventListener('resize', this._reposition);
+		window.removeEventListener('scroll', this._reposition);
+	}
+
+	render() {
+		return html`
+			<div class=${classMap(this._containerClasses)} >
+				<div class="d2l-floating-buttons-inner-container"
+					style=${styleMap(this._innerContainerStyle)}>
+					<slot></slot>
+				</div>
+			</div>
+			<div class="d2l-floating-buttons-spacer"
+				style=${styleMap(this._buttonSpacerStyle)}>
+			</div>
+		`;
+	}
+
+	firstUpdated() {
+		this._container = this.shadowRoot.querySelector('.d2l-floating-buttons-container');
+		this._spacer = this.shadowRoot.querySelector('.d2l-floating-buttons-spacer');
+		this.updateComplete.then(() => {
+			this._reposition();
+			let prevDocumentHeight = document.body.offsetHeight;
+			setInterval(() => {
+				const documentHeight = document.body.offsetHeight;
+				if (prevDocumentHeight !== documentHeight) {
+					this._reposition();
+				}
+				prevDocumentHeight = documentHeight;
+			}, 100);
+		});
+	}
+
+	/**
+	 * Whether or not the buttons are floating.
+	 */
+	isFloating() {
+		return this._floatingButtonsFloating;
+	}
+
+	_reposition() {
+		if (this.minHeight) {
+			this._viewportIsAtLeastMinHeight = window.matchMedia(`(max-height: ${this.minHeight})`).matches;
```

Comment:
does this need to be a property since it only seems to be used in `_reposition`?